File: python/007/main.py
# ProjectEuler 007
# Python / main.py
# copyright (c) 2015 Susisu
#
# Python 3.4.2
# `python3 main.py`

from math import sqrt, floor

def main():
    counter = 0
    n = 1
    while counter < 10001:
        n += 1
        if isPrime(n):
            counter += 1
    print(n)

def isPrime(n):
    if n < 2:
        return False
    else:
        for i in range(2, floor(sqrt(n)) + 1):
            if n % i == 0:
                return False
        return True

main()

File: perl/007/main.pl
# ProjectEuler 007
# Perl / main.pl
# copyright (c) 2015 Susisu
#
# Perl 5.16.2
# `perl main.pl`

sub isPrime {
    my ($n) = @_;
    if ($n < 2) {
        return 0;
    }
    else {
        my $r = sqrt $n;
        for (my $i = 2; $i <= $r; $i++) {
            if ($n % $i == 0) {
                return 0;
            }
        }
        return 1;
    }
}

sub main {
    my $c = 0;
    my $n = 1;
    while ($c < 10001) {
        $n++;
        if (isPrime $n) {
            $c++;
        }
    }
    print $n . "\n";
}

&main;
